frontend/src/metabase/metadata/components/NameDescriptionInput/NameDescriptionInput.tsx

```diff
@@ -22,20 +22,28 @@ export const NameDescriptionInput = ({
 }: Props) => {
   return (
     <Box>
-      <TextInputBlurChange
+      <Input
         classNames={{
           input: S.nameInput,
           root: S.name,
         }}
         fw="bold"
+        normalize={(newValue) => {
+          if (typeof newValue !== "string") {
+            return name;
+          }
+
+          const isNewValueEmpty = newValue.trim().length === 0;
+          return isNewValueEmpty ? name : newValue.trim();
+        }}
         placeholder={namePlaceholder}
         required
         size="lg"
         value={name}
         onChange={onNameChange}
       />
 
-      <TextInputBlurChange
+      <Input
```

Comment:
I managed to reuse existing `TextInputBlurChange`. `Input` is still a thing due to UX/styling reasons.